### PR TITLE
Add APIType, enums for server names and util to generate (local) OverkizServer class

### PR DESCRIPTION
**Pull request by iMicknl**
Adds an enum + variable to the client class to make it easier to detect if local or cloud API is used.

### CODE REVIEW

File: pyoverkiz/client.py [hunk 1]
```diff
@@ -129,6 +131,12 @@ def __init__(
 
         self.session = session if session else ClientSession()
 
+        self.api_type = (
```
Comment (tetienne) [inline pyoverkiz/client.py:134]: I'm not a big fan of this style.

Comment (iMicknl) [reply pyoverkiz/client.py:134]: What would you prefer? Just a regular if/else statement?

**Comment by iMicknl** [comment]
@tetienne I am thinking about porting this as well: https://github.com/home-assistant/core/blob/50ae1d8e2735ade85cb68e08d9022fc6ac5bf899/homeassistant/components/overkiz/config_flow.py#L39-L55, to have more code in our client than in the integration.

File: pyoverkiz/enums/server.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+@unique
+class APIType(str, Enum):
+    CLOUD = "cloud"
+    LOCAL = "local"
+
+
+@unique
+class Server(str, Enum):
```
Comment (iMicknl) [inline pyoverkiz/enums/server.py:11]: @tetienne I already added this one from your PR. Is a great add and would be good to have this in soon, before we fully migrate to the "v2" style of this client.

File: pyoverkiz/client.py [hunk 3]
```diff
@@ -129,6 +131,11 @@ def __init__(
 
         self.session = session if session else ClientSession()
 
+        if "/enduser-mobile-web/1/enduserAPI/" in self.server.endpoint:
```
Comment (tetienne) [inline pyoverkiz/client.py:134]: Can you create a constant?

File: pyoverkiz/const.py [hunk 4]
```diff
@@ -16,86 +17,92 @@
 SOMFY_CLIENT_ID = "0d8e920c-1478-11e7-a377-02dd59bd3041_1ewvaqmclfogo4kcsoo0c8k4kso884owg08sg8c40sk4go4ksg"
 SOMFY_CLIENT_SECRET = "12k73w1n540g8o4cokg0cw84cog840k84cwggscwg884004kgk"
 
+SERVERS_THAT_SUPPORT_LOCAL_API = [
```
Comment (tetienne) [inline pyoverkiz/const.py:20]: What about?

```suggestion
SERVERS_W_LOCAL_API = [
```

File: pyoverkiz/enums/server.py (hunk 2 above)
Comment (tetienne) [reply pyoverkiz/enums/server.py:11]: Perfect, I will see if I can backport more changes without breaking anything

File: pyoverkiz/const.py (hunk 4 above)
Comment (iMicknl) [reply pyoverkiz/const.py:20]: I am not a huge fan of abbreviations like `W`, what about `SERVERS_WITH_LOCAL_API`? Or `DEV_MODE_SERVERS`?

File: pyoverkiz/utils.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    """Generate OverkizServer class for connection with a local API (Somfy Developer mode)."""
+    return OverkizServer(
+        name=name,
+        endpoint=f"https://{host}{LOCAL_API_PATH}",
+        manufacturer=manufacturer,
+        configuration_url=configuration_url,
+    )
+
+
+def is_overkiz_gateway(gateway_id: str) -> bool:
```
Comment (tetienne) [inline pyoverkiz/utils.py:24]: What’s an additional gateway?

**Review by tetienne — APPROVED** [review]

File: pyoverkiz/utils.py (hunk 5 above)
Comment (iMicknl) [reply pyoverkiz/utils.py:24]: Some users have a thermostat + protect system, which is also listed as a gateway. We have a JSON definition of one of them in our fixtures folder.

**Pull request merged**